smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/endpoints/EndpointResolverGenerator.kt

```diff
@@ -9,111 +9,59 @@ import software.amazon.smithy.codegen.core.Symbol
+        writer.write(
+            "typealias DefaultEndpointResolver = \$N<EndpointParams>",
+            ClientRuntimeTypes.Core.DefaultEndpointResolver,
+        )
+        writer.write("")
+        writer.openBlock("extension DefaultEndpointResolver {", "}") {
+            endpointRuleSet?.let {
+                writer.write("private static let ruleSet = \$S", Node.printJson(it.toNode()))
+            } ?: run {
+                writer.write("private static let ruleSet = \"{}\"")
             }
             writer.write("")
-            writer.openBlock(
-                "public func resolve(params: EndpointParams) throws -> \$L {", "}", SmithyHTTPAPITypes.Endpoint
-            ) {
-                endpointRules?.let {
-                    writer.write("let context = try \$L()", ClientRuntimeTypes.Core.EndpointsRequestContext)
-                    endpointRules.parameters?.toList()?.sortedBy { it.name.toString() }?.let { sortedParameters ->
-                        sortedParameters.forEach { param ->
-                            val memberName = param.name.toString().toLowerCamelCase()
-                            val paramName = param.name.toString()
-                            writer.write("try context.add(name: \$S, value: params.\$L)", paramName, memberName)
-                        }
-                        writer.write("")
-                    }
-                    writer.openBlock("guard let crtResolvedEndpoint = try engine.resolve(context: context) else {", "}") {
-                        writer.write("throw EndpointError.unresolved(\"Failed to resolved endpoint\")")
-                    }.write("")
-
-                    writer.openBlock("if crtResolvedEndpoint.getType() == .error {", "}") {
-                        writer.write("let error = crtResolvedEndpoint.getError()")
-                        writer.write("throw EndpointError.unresolved(error)")
-                    }.write("")
-
-                    writer.openBlock("guard let url = crtResolvedEndpoint.getURL() else {", "}") {
-                        writer.write("assertionFailure(\"This must be a bug in either CRT or the rule engine, if the endpoint is not an error, it must have a url\")")
-                        writer.write("throw EndpointError.unresolved(\"Failed to resolved endpoint\")")
-                    }.write("")
-
-                    writer.write("let headers = crtResolvedEndpoint.getHeaders() ?? [:]")
-                    writer.write("let properties = crtResolvedEndpoint.getProperties() ?? [:]")
-                    writer.write("return try Endpoint(urlString: url, headers: Headers(headers), properties: properties)")
-                } ?: run {
-                    writer.write("fatalError(\"EndpointResolver not implemented\")")
-                }
+            writer.openBlock("init() throws {", "}") {
+                writer.write("try self.init(partitions: \$N, ruleSet: Self.ruleSet)", partitionDefinition)
```

Comment:
This convenience initializer creates the endpoint resolver with the rule set & partitions pre-filled.